kivy/uix/floatlayout.py

```diff
@@ -78,12 +78,17 @@ def do_layout(self, *largs, **kwargs):
         for c in self.children:
             # size
             shw, shh = c.size_hint
-            if shw and shh:
-                c.size = w * shw, h * shh
-            elif shw:
-                c.width = w * shw
-            elif shh:
-                c.height = h * shh
+            if shw is not None:
+                local_width = w * shw
+            else:
+                local_width = c.width
+
+            if shh is not None:
+                local_height = h * shh
+            else:
+                local_height = c.height
+
+            c.size = local_width, local_height
```

Comment:
You're reassigning the child's width/height to itself when the hint is None. That's very inefficient because the property gets re-evaluated. You should probably do it like before but just replace e.g. `if shw and shh:` with `if shw is not None and shh is not None:`.